docs/advanced-guides/testing/testing-with-react-testing-library.md

```diff
@@ -1,13 +1,117 @@
-# Testing With React Testing Library
+# Testing With React Testing Library (RTL)
 
 ## Getting Setup
 
-TODO
+This guide assumes you followed the instructions for [Adding React Router via Create React App](../../installation/add-to-a-website.md#create-react-app). If you did not start with Create React App, but still have the same application structure, you can install [React Testing Library](https://testing-library.com/docs/react-testing-library/intro) by following [their installation guide](https://github.com/testing-library/react-testing-library#installation).
+
+## Basic Test
+
+A basic rendering test can be important to esnure that we have everything installed and set up correctly. Fortunately, this is easy to do with RTL.
+
+Since we've wrapped our `App` component in the `Router` in our `index.js` file, we do have to wrap it in each of our isolated component tests, otherwise `history` will be undefined. If the `Router` had been inside of our `App`, then we would not have to wrap it inside of our tests.
+
+A recommended simple test looks like the following:
+
+```jsx
+test('renders react router header', () => {
+  const { getByText } = render(<Router><App /></Router>);
```

Comment:
We're presently recommending people import `screen` from RTL instead of destructuring the render result. This works better with autocomplete and makes each test a little simpler.

Reply:
Yup, so this would be:

```javascript
import {screen, render} from '@testing-library/react';

test('renders react router header', () => {
  render(<Router><App /></Router>);
  const header = screen.getByText('Welcome to React Router!');
  expect(header).toBeInTheDocument();
})
```

It would be even better to recommend the `wrapper` option so `rerender` works as expected:

```javascript
import {screen, render} from '@testing-library/react';

test('renders react router header', () => {
  render(<App />, {wrapper: Router});
  const header = screen.getByText('Welcome to React Router!');
  expect(header).toBeInTheDocument();
})
```

And then to take it to the next level, there's a custom render: https://testing-library.com/docs/react-testing-library/setup#custom-render

Hope that context is helpful!

Reply:
I see that in the repo README @alexkrolick but not in the [docs site](https://testing-library.com/docs/react-testing-library/example-intro), the destructuring is still there, which is what I've been going off of for a while.

I can definitely update these examples, and I can also put in a PR for the RTL docs.

Reply:
Sorry, we're still in the process of updating things to the new recommendations. PRs would be appreciated!

Reply:
No worries, I can definitely help with that.

Thanks for all the help with this! I'll work on some updates for this tonight.